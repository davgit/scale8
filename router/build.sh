#!/bin/bash

# Clean up...
rm -rf ../ui/build ../ui/out

yarn --cwd ../ui export

rsync -a --delete ../ui/out/ ui-build

docker build -t scale8/router:latest .
