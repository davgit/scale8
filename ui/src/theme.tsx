import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
