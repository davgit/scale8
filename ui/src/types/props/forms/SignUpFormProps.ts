import { FormProps } from '../../../hooks/form/useFormValidation';
import { ApolloError } from '@apollo/client';
import { RefObject } from 'react';
import HCaptcha from '@hcaptcha/react-hcaptcha';

export type SignUpValues = {
    newPassword: string;
    newPasswordConfirm: string;
    domain: string;
    orgName: string;
    fullName: string;
    email: string;
    agree: boolean;
    CAPTCHAToken: string;
    tempAccessCode: string;
};

export type SignUpFormProps = FormProps<SignUpValues> & {
    gqlError?: ApolloError;
    success: boolean;
    email?: string;
    type: string;
    target: string | undefined;
    loading: boolean;
    qsEmail: string | undefined;
    captcha: RefObject<HCaptcha>;
};
