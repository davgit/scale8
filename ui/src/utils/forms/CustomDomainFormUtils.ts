import { ValidateConfiguration } from '../validators/validateFormValues';
import domainValidator from '../validators/domainValidator';
import { FormProps } from '../../hooks/form/useFormValidation';

export type CustomDomainValues = {
    domain: string;
    certificate: string;
    key: string;
};

export const CustomDomainValidators: ValidateConfiguration<CustomDomainValues>[] = [
    {
        field: 'domain',
        validator: domainValidator,
        error: () => 'Invalid domain',
        ignoreEmpty: true,
    },
];

export type CustomDomainFormProps = FormProps<CustomDomainValues> & {
    cname: string;
};
