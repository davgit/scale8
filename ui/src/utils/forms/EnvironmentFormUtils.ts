import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import urlValidator from '../validators/urlValidator';

export type EnvVariable = { key: string; value: string };

export type EnvironmentValues = {
    name: string;
    url: string;
    revisionId: string;
    variables?: EnvVariable[];
    comments: string;
};

const EnvironmentCreateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

const EnvironmentUpdateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

export { EnvironmentCreateValidators, EnvironmentUpdateValidators };
