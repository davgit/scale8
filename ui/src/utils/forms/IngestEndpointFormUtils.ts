import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import { StorageProviderFields, storageProviderValidators } from '../StorageProviderUtils';
import { IngestSchemaWizard } from '../../gql/generated/globalTypes';

export type IngestEndpointValues = StorageProviderFields & {
    name: string;
    wizard: IngestSchemaWizard | '';
    analyticsEnabled: boolean;
};

const IngestEndpointValidators: ValidateConfiguration<IngestEndpointValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Ingest Endpoint name too short',
    },
    ...storageProviderValidators,
];

export { IngestEndpointValidators };
