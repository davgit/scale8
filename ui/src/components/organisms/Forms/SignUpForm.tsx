import { FC, useEffect, useState } from 'react';
import {
    Box,
    Button,
    Checkbox,
    FormControl,
    FormControlLabel,
    FormGroup,
    FormHelperText,
} from '@mui/material';
import FormError from '../../atoms/FormError';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import SignUpContainer from '../../molecules/SignUpContainer';
import Loader from '../Loader';
import Captcha from '../../atoms/Captcha';
import { SignUpFormProps } from '../../../types/props/forms/SignUpFormProps';
import Link from '../../atoms/Next/Link';
import FormFull from '../../atoms/FormFull';

const SignUpForm: FC<SignUpFormProps> = (props: SignUpFormProps) => {
    const { type, values, handleChange, errors, qsEmail } = props;

    const setCaptchaToken = (token: string) => handleChange('CAPTCHAToken', token);

    const determineSubmitClass = () => {
        if (type === 'tag-manager') {
            return 'formFullTMColorSubmit';
        }

        if (type === 'data-manager') {
            return 'formFullDMColorSubmit';
        }

        return 'formFullMainColorSubmit';
    };

    const [generatePassword, setGeneratePassword] = useState(true);

    useEffect(() => {
        if (values.newPassword !== '' && values.newPasswordConfirm !== '' && generatePassword) {
            handleChange('newPassword', '', [{ valueKey: 'newPasswordConfirm', value: '' }]);
        }
    }, [values, handleChange, generatePassword]);

    if (props.loading) {
        return (
            <SignUpContainer
                type={props.type}
                target={props.target}
                isCompleted={false}
                isPrepare={false}
            >
                <Loader />
            </SignUpContainer>
        );
    }

    if (props.success) {
        return (
            <SignUpContainer
                type={props.type}
                target={props.target}
                isCompleted={true}
                isPrepare={false}
            >
                <Box mb={2} width="100%">
                    <Box py={10}>
                        <Box fontSize={18} width="100%" textAlign="center">
                            Please <b>click the link in the email</b> we have just sent to{' '}
                            {props.email}.
                        </Box>
                        <Box pt={1} />
                        <Box fontSize={18} width="100%" textAlign="center">
                            Could this have gone to your spam folder?
                        </Box>
                    </Box>
                </Box>
            </SignUpContainer>
        );
    }

    return (
        <SignUpContainer
            type={props.type}
            target={props.target}
            isCompleted={false}
            isPrepare={false}
        >
            {props.gqlError && (
                <Box my={2} width="100%">
                    <FormError error={props.gqlError.message} />
                </Box>
            )}
            <FormFull handleSubmit={props.handleSubmit}>
                <ControlledTextInput
                    name="fullName"
                    label="Full Name"
                    formProps={props}
                    variant="outlined"
                    margin="normal"
                    required
                    fullWidth
                />
                {type !== 'invite' && qsEmail === undefined && (
                    <ControlledTextInput
                        name="email"
                        label="Email Address"
                        formProps={props}
                        variant="outlined"
                        margin="normal"
                        required
                        fullWidth
                    />
                )}
                {type !== 'invite' && (
                    <ControlledTextInput
                        name="orgName"
                        label="Organization name"
                        formProps={props}
                        variant="outlined"
                        margin="normal"
                        required
                        fullWidth
                    />
                )}
                {type === 'tag-manager' && (
                    <ControlledTextInput
                        name="domain"
                        label="Domain"
                        formProps={props}
                        variant="outlined"
                        margin="normal"
                        required
                        fullWidth
                    />
                )}
                {!generatePassword && (
                    <>
                        <ControlledTextInput
                            name="newPassword"
                            label="New Password"
                            formProps={props}
                            variant="outlined"
                            margin="normal"
                            required
                            fullWidth
                            inputProps={{
                                autoComplete: 'new-password',
                            }}
                            type="password"
                        />
                        <ControlledTextInput
                            name="newPasswordConfirm"
                            label="Confirm New Password"
                            formProps={props}
                            variant="outlined"
                            margin="normal"
                            required
                            fullWidth
                            inputProps={{
                                autoComplete: 'new-password',
                            }}
                            type="password"
                        />
                    </>
                )}
                <FormGroup>
                    <FormControlLabel
                        control={
                            <Checkbox
                                checked={generatePassword}
                                onChange={() => {
                                    setGeneratePassword(!generatePassword);
                                }}
                                size="medium"
                                name="generatePassword"
                            />
                        }
                        label="Auto-generate secure password"
                    />
                </FormGroup>
                <FormControl error={errors['agree'] !== undefined}>
                    <FormGroup>
                        <FormControlLabel
                            sx={{
                                color: (theme) =>
                                    errors['agree'] === undefined
                                        ? undefined
                                        : theme.palette.error.main,
                            }}
                            control={
                                <Checkbox
                                    checked={values.agree}
                                    onChange={() => handleChange('agree', !values.agree)}
                                    size="medium"
                                    name="agree"
                                />
                            }
                            label={
                                <Box
                                    component="span"
                                    sx={{
                                        color: (theme) =>
                                            errors['agree'] === undefined
                                                ? undefined
                                                : theme.palette.error.main,
                                    }}
                                >
                                    I agree to the{' '}
                                    <Link
                                        href={'https://scale8.com/legal/terms'}
                                        sx={{ color: 'rgba(0, 0, 0, 0.87)' }}
                                    >
                                        Terms of Service
                                    </Link>
                                </Box>
                            }
                        />
                    </FormGroup>
                    {errors['agree'] !== undefined && (
                        <FormHelperText>{errors['agree']}</FormHelperText>
                    )}
                </FormControl>
                <Box mt={2} />
                <FormControl error={errors['CAPTCHAToken'] !== undefined}>
                    <Captcha captcha={props.captcha} setCaptchaToken={setCaptchaToken} />
                    {errors['CAPTCHAToken'] !== undefined && (
                        <FormHelperText>{errors['CAPTCHAToken']}</FormHelperText>
                    )}
                </FormControl>
                <Button
                    type="submit"
                    fullWidth
                    variant="contained"
                    className={determineSubmitClass()}
                    disabled={props.isSubmitting}
                >
                    {props.submitText}
                </Button>
            </FormFull>
        </SignUpContainer>
    );
};

export default SignUpForm;
