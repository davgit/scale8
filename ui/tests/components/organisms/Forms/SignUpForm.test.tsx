import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { createRef, FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import SignUpForm from '../../../../src/components/organisms/Forms/SignUpForm';
import { SignUpValues } from '../../../../src/types/props/forms/SignUpFormProps';
import HCaptcha from '@hcaptcha/react-hcaptcha';
import * as nextRouter from 'next/router';

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
nextRouter.useRouter = jest.fn();
(nextRouter.useRouter as jest.Mock).mockImplementation(() => ({ route: '/' }));

const MockForm: FC = () => {
    const formProps = useMockFormProps<SignUpValues>({
        newPassword: '',
        newPasswordConfirm: '',
        domain: '',
        orgName: '',
        fullName: '',
        email: '',
        agree: false,
        CAPTCHAToken: '',
    });

    const captcha = createRef<HCaptcha>();

    return (
        <SignUpForm
            target={undefined}
            qsEmail={undefined}
            captcha={captcha}
            success={false}
            type={'tag-manager'}
            loading={false}
            {...formProps}
        />
    );
};

describe('SignUpForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
