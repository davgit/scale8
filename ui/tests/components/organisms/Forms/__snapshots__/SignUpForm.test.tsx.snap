// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm form render unchanged 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-cegarq"
    >
      <svg
        height="151"
        viewBox="0 0 29.036954 38.838082"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 0 9.721 C 0 4.3775 4.332 0.0458 9.675 0.0458 L 9.675 9.721 Z"
          id="path158"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
        <path
          d="M 29.037 0.0468 C 29.037 5.3903 24.705 9.7221 19.361 9.7221 L 19.361 0.0468 Z"
          id="path162"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
        <path
          d="M 19.361 29.1625 C 19.361 34.506 15.029 38.8378 9.686 38.8378 L 9.686 29.1625 Z"
          id="path166"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
        <path
          d="M 9.675 0.0002 L 19.361 0.0002 L 19.361 9.7221 L 9.675 9.7221 Z"
          id="path168"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
        <path
          d="M 9.675 9.721 L 19.361 9.721 L 19.361 19.4418 L 9.675 19.4418 Z"
          id="path170"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
        <path
          d="M 9.675 19.4418 L 19.361 19.4418 L 19.361 29.1636 L 9.675 29.1636 Z"
          id="path172"
          style="fill: #39cce0; fill-opacity: 1; fill-rule: nonzero; stroke: none;"
        />
      </svg>
    </div>
    <div
      class="MuiBox-root css-19ayr1g"
    >
      Sign-up for an unlimited 30 day free trial.
    </div>
    <div
      class="MuiBox-root css-zoser8"
    />
    <div
      class="MuiBox-root css-1181e6k"
    >
      No payment information or credit card is required.
    </div>
    <div
      class="MuiBox-root css-1v3caum"
    />
    <div
      class="MuiBox-root css-hiapv6"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 MuiCard-root css-1sxjyu4-MuiPaper-root-MuiCard-root"
      >
        <div>
          <div
            class="MuiStepper-root MuiStepper-horizontal css-m5vj9m-MuiStepper-root"
            color="red"
          >
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal css-1fi72el-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1hbon82-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-pj2ghc-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      1
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-1ie2zp2-MuiStepLabel-label"
                  >
                    Create Account
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-1fi72el-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1hbon82-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-pj2ghc-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      2
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1ie2zp2-MuiStepLabel-label"
                  >
                    Confirm Email
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-1fi72el-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1hbon82-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-pj2ghc-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      3
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1ie2zp2-MuiStepLabel-label"
                  >
                    Prepare Account
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-1bw0nnu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-1fi72el-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1hbon82-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-pj2ghc-MuiStepIcon-text"
                      text-anchor="middle"
                      x="12"
                      y="16"
                    >
                      4
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1ie2zp2-MuiStepLabel-label"
                  >
                    Install Tags
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1yuhvjn"
        />
        <form
          class="MuiBox-root css-1u15dwc"
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              Full Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-dloxrs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                name="fullName"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Full Name *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-2"
              id="mui-2-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-dloxrs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-2"
                name="email"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Email Address *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-3"
              id="mui-3-label"
            >
              Organization name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-dloxrs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-3"
                name="orgName"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Organization name *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-4"
              id="mui-4-label"
            >
              Domain
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-dloxrs-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-4"
                name="domain"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Domain *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  name="generatePassword"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
              >
                Auto-generate secure password
              </span>
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div
              class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
            >
              <label
                class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
              >
                <span
                  class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    name="agree"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
                >
                  <span
                    class="MuiBox-root css-0"
                  >
                    I agree to the 
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-zlr2jj-MuiTypography-root-MuiLink-root"
                      href="https://scale8.com/legal/terms"
                    >
                      Terms of Service
                    </a>
                  </span>
                </span>
              </label>
            </div>
          </div>
          <div
            class="MuiBox-root css-1yuhvjn"
          />
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div />
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root formFullTMColorSubmit css-8gt83u-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
